test/python/qpy/test_circuit_load_from_qpy.py

```diff
@@ -185,3 +193,17 @@ def test_no_register(self, opt_level):
             list(new_circuit.layout.input_qubit_mapping.values()),
         )
         self.assertEqual(tqc.layout.final_layout, new_circuit.layout.final_layout)
+
+    def test_invalid_version_value(self):
+        """Assert we raise an error with an invalid version request."""
+        qc = QuantumCircuit(2)
+        with self.assertRaises(ValueError):
+            dump(qc, io.BytesIO(), version=3)
+
+    def test_version_10_roundtrip(self):
+        """Test the version is set correctly when specified."""
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.measure_all()
+        self.assert_roundtrip_equal(qc, version=10)
```

Comment:
You could stop hardcoding 10 here (but besides my code suggestion you also need to import `QPY_COMPATIBILITY_VERSION`):

```suggestion
    def test_compatibility_version_roundtrip(self):
        """Test the version is set correctly when specified."""
        qc = QuantumCircuit(2)
        qc.h(0)
        qc.cx(0, 1)
        qc.measure_all()
        self.assert_roundtrip_equal(qc, version=QPY_COMPATIBILITY_VERSION)
```

Reply:
Updated in: https://github.com/Qiskit/qiskit/pull/11644/commits/b83c197e2df94d52ec9d71418c1f33ac8089057c